docs/root/adaptive_load_controller.md

```diff
@@ -0,0 +1,202 @@
+[scoring_function_impl.cc](https://github.com/envoyproxy/nighthawk/blob/main/source/adaptive_load/scoring_function_impl.cc)
+file.
+
+Scoring function | Plugin Name                | Description
+---------------- | -------------------------- | -----------
+Binary scoring   | `nighthawk.binary_scoring` | A boolean (true/false) evaluation of the metric value.
+Linear scoring   | `nighthawk.linear_scoring` | Evaluation of the metric value with a linear scoring constant.
+
+### Available input variable setters
+
+The existing input variable setters are implemented in the
+[input_variable_setter_impl.cc](https://github.com/envoyproxy/nighthawk/blob/main/source/adaptive_load/input_variable_setter_impl.cc)
+file.
+
+Input variable setter | Plugin Name     | Description
+--------------------- | --------------- | -----------
+RPS                   | `nighthawk.rps` | Sets the RPS value in the Nighthawk configuration for an iteration.
+
+## Configuration
+
+The controller accepts its
+[configuration](https://github.com/envoyproxy/nighthawk/blob/main/api/adaptive_load/adaptive_load.proto)
+in protocol buffer format.
+
+A valid configuration contains a `nighthawk_traffic_template` which is the base
+configuration for Nighthawk to be used in the iterations. The chosen input
+variable setter alters variables in this template.
+
+The configuration also contains a selection of plugins, their individual
+configuration and a set of timers that determine the duration and deadline of
+the individual stages used in the adaptive mode.
+
+The following outlines an example configuration for the adaptive mode that uses
+the exponential search algorithm to find an RPS value that achieves more than
+99% of the `send-rate` metric:
+
+```
+metric_thresholds {
+  metric_spec { metric_name: "send-rate" }
+  threshold_spec {
+    scoring_function {
+      name: "nighthawk.binary_scoring"
+      typed_config {
+        [type.googleapis.com/nighthawk.adaptive_load.BinaryScoringFunctionConfig] {
+          lower_threshold { value: 0.99 }
+        }
+      }
+    }
+  }
+}
+
+step_controller_config {
+  name: "nighthawk.exponential_search"
+  typed_config {
+    [type.googleapis.com/nighthawk.adaptive_load.ExponentialSearchStepControllerConfig] {
+      initial_value: 200
+      input_variable_setter {
+        typed_config {
+          [type.googleapis.com/nighthawk.adaptive_load.RequestsPerSecondInputVariableSetterConfig] { }
+        }
+      }
+    }
+  }
+}
+
+measuring_period { seconds: 30 }
+
+benchmark_cooldown_duration { seconds: 60 }
+
+convergence_deadline { seconds: 1200 }
+
+testing_stage_duration { seconds: 60 }
+```
+
+It is recommended to use configuration that contains at least two metric
+thresholds to ensure stable test results. A typical adaptive mode configuration
+will contain the following metric thresholds:
+
+*  Send rate to ensure Nighthawk has enough resources to deliver the load.
+*  Success rate to ensure the System under test is responding.
```

Comment:
system

Reply:
Done in 87ce69b6093b33651f29dd3b2e6b2373527c26cd.